### PR TITLE
Add Dataset Module field to DatasetInfo

### PR DESCRIPTION
**Pull request by NikhilBartwal**
In reference to Issue #2813 
As suggested by @Conchylicultor, a new `module_field` was added to the `dataset_info` to hold the original module location of the dataset.
The added change was also tested by updating the `INFO_STR` in `dataset_info_test` and then comparing it with the new ly generated `dataset_info`. 

The `mnist dataset_info.json` file used for testing was also updated manually to contain module_name so as it can be tested for `assertEqual` with the newly generated one.
The code performed as expected without generating any failures (except some which were present previous to the changes)

### CODE REVIEW

**Comment by NikhilBartwal** [comment]
@Conchylicultor Could you have a look if this is indeed what was needed? Thanks!

**Review by Conchylicultor — APPROVED** [review]
Thank you. Looks good

**Comment by NikhilBartwal** [comment]
@Conchylicultor There are a couple more `AssertionErrors` that occur when `dataset_info_test.py` is ran. Should i open separate issues for that or should i wait for this to get merged first?

**Comment by Conchylicultor** [comment]
`test_utils.py` should not be executed as test: https://github.com/tensorflow/datasets/blob/8ee49991924a058b91b410f3bc4309f3a764faa2/oss_scripts/oss_tests.sh#L45

I've imported the code internally, so no need to fix anything from your side!

**Comment by NikhilBartwal** [comment]
@Conchylicultor I ran `dataset_info_test.py` to test whether it is generated correctly (the same script was used to test this PR as well). `test_utils.py` was not run directly but was a part of `dataset_into_test.py`.

This was the test which resulted in the mentioned assertion errors. 
The mentioned code was only a part of the stack trace.

File: tensorflow_datasets/core/proto/dataset_info_generated_pb2.py [hunk 1]
```diff
@@ -37,7 +35,8 @@
...
+  create_key=_descriptor._internal_create_key,
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/proto/dataset_info_generated_pb2.py:38]: Could you share which protobuf version you used to generate the `_pb2.py` ? I think the new version might break users which are still using an older protobuf package version: https://stackoverflow.com/questions/61922334/how-to-solve-attributeerror-module-google-protobuf-descriptor-has-no-attribu

Could you try to generate the proto with an older protobuf version which avoid the `_internal_create_key` ?

**Review by Conchylicultor — CHANGES_REQUESTED** [review]

File: tensorflow_datasets/core/proto/dataset_info_generated_pb2.py (hunk 1 above)
Comment (NikhilBartwal) [reply tensorflow_datasets/core/proto/dataset_info_generated_pb2.py:38]: @Conchylicultor I used the v3.15 by custom download and setup as the v3.11 provided for download as part of the `install_protoc.sh` was not compatible for windows.

I will revert back to an older version (preferably 3.11) and try generating the proto again!

**Comment by Conchylicultor** [comment]
> @Conchylicultor I ran `dataset_info_test.py` to test whether it is generated correctly (the same script was used to test this PR as well). `test_utils.py` was not run directly but was a part of `dataset_into_test.py`.
> 
> This was the test which resulted in the mentioned assertion errors.
> The mentioned code was only a part of the stack trace.

For your issues, I believe this is because you haven't installed required extra-tests deps like `tensorflow-data-validation`. See https://www.tensorflow.org/datasets/contribute#cloning_the_repo

**Comment by NikhilBartwal** [comment]
@Conchylicultor The `_internal_create_key` issue was fixed by reverting back to an older `v3.11.2` and regenerating the proto. The required tests passed successfully.

**Comment by NikhilBartwal** [comment]
> 
> For your issues, I believe this is because you haven't installed required extra-tests deps like `tensorflow-data-validation`. See https://www.tensorflow.org/datasets/contribute#cloning_the_repo
> 
> ```
> pip install -e .[dev]
> ```
@Conchylicultor My bad, i didn't know about that. Thanks for clearing that out!

**Review by Conchylicultor — APPROVED** [review]
Awesome! Merging this soon

**Comment by vijayphoenix** [comment]
Merged by https://github.com/tensorflow/datasets/commit/0a1875c7f43a48932225dca707c1decd7031c7b3

**Pull request closed without merging**